### PR TITLE
add upload a contract

### PR DESCRIPTION
**Pull request by BlaiseJavan**
#### What does this PR do?

- Type here what your pr will do in short

#### Description of Task to be completed?

- Name all the task completed in reference to the one listed in the Jira description

  - [] Task One

  - [] Task Two

#### How should this be manually tested?

- Clone the project through your terminal(VScode terminal)

      git clone https://github.com/Volve-gitore/quickss-backend.git

- Go to the project and Open the project

       cd quickss-backend && cd .

- Open the feature through its branch

       - git checkout 'branch name'

- Install all the dependencies

       npm i

#### Any background context you want to provide?

- Provide some description/context one should know about your PR

#### What are the relevant Jira story?

- add the link got from Jira

#### Screenshots (if appropriate)

- Add some Image if needed

### CODE REVIEW

File: src/controllers/Client.js [hunk 1]
```diff
@@ -2,39 +2,69 @@ import model from '../db/models';
...
       // get all images from the request
+      // for (let image in req.files) {
+      //   const dataBuffer = new Buffer.from(req.files[image].buffer);
+      //   const mediaType = path.extname(req.files[image].originalname).toString();
+
+      //   const imageData = imageDataURI.encode(dataBuffer, mediaType);
+      //   const uploadedImage = await uploader.upload(imageData);
+      //   images.push(uploadedImage.url);
+      // }
+
```
Comment (Emile-Nsengimana) [inline src/controllers/Client.js:32]: remove this

File: src/controllers/Client.js [hunk 2]
```diff
@@ -2,39 +2,69 @@ import model from '../db/models';
...
 
+      const file = req.files.contract[0];
+      const fileNamee = Date.now()+""+file.originalname;
+      const dataBuffer = new Buffer.from(file.buffer);
+      const mediaType = path.extname(file.originalname).toString();
+
+      const a = imageDataURI.encode(dataBuffer, mediaType);
+
+      const abc = await fs.writeFileSync(
+        '/home/blaise/Documents/projects/volve/quickss/quickss-backend/uploads/'+fileNamee,
```
Comment (Emile-Nsengimana) [inline src/controllers/Client.js:50]: Try to avoid hard-coding the `path` to upload files

File: src/controllers/Client.js [hunk 3]
```diff
@@ -2,39 +2,69 @@ import model from '../db/models';
...
       }
 
+      const file = req.files.contract[0];
+      const fileNamee = Date.now()+""+file.originalname;
+      const dataBuffer = new Buffer.from(file.buffer);
+      const mediaType = path.extname(file.originalname).toString();
+
+      const a = imageDataURI.encode(dataBuffer, mediaType);
+
+      const abc = await fs.writeFileSync(
```
Comment (Emile-Nsengimana) [inline src/controllers/Client.js:49]: Use meaningful variable name

File: src/controllers/Client.js [hunk 4]
```diff
@@ -2,39 +2,69 @@ import model from '../db/models';
...
       // save client's information to database
-      const client = await Client.create({ ...req.body, images });
+      const client = await Client.create({ ...req.body, images, contract });
 
       if (client)
         return res.status(201).send({
           message: `${req.body.category} added successful`,
         });
     } catch (error) {
+      console.log('error ', error);
```
Comment (Emile-Nsengimana) [inline src/controllers/Client.js:67]: remove all console

File: src/controllers/Client.js [hunk 5]
```diff
@@ -2,39 +2,69 @@ import model from '../db/models';
...
+        const dataBuffer = new Buffer.from(req.files.image[0].buffer);
+        const mediaType = path.extname(req.files.image[0].originalname).toString();
 
         const imageData = imageDataURI.encode(dataBuffer, mediaType);
         const uploadedImage = await uploader.upload(imageData);
         images.push(uploadedImage.url);
       }
 
+      const file = req.files.contract[0];
+      const fileNamee = Date.now()+""+file.originalname;
```
Comment (Emile-Nsengimana) [inline src/controllers/Client.js:43]: It will be better to use `date-clientname-contract` as contract file name format

**Review by Emile-Nsengimana — CHANGES_REQUESTED** [review]
Check all comments above and also remove all `pdf` files used when testing

File: src/controllers/Client.js (hunk 1 above)
Comment (BlaiseJavan) [reply src/controllers/Client.js]: done

File: src/controllers/Client.js (hunk 2 above)
Comment (BlaiseJavan) [reply src/controllers/Client.js]: all cleared

**Review by Emile-Nsengimana — APPROVED** [review]

**Pull request merged**